### DISCUSSION TITLE
Webhook Support?

### DISCUSSION THREAD

**Discussion by billyjacoby** [blakeblackshear/frigate#8134] [Ask A Question]
I've looked through the codebase and haven't been able to find anything, but I'm hoping that I'm missing something.

Is there any sort of webhook support that Frigate offers out of the box? My main use case is for to send notifications a server that could then notify mobile devices via a native app that I'm building.

If there isn't support for this yet, is it something that a PR would be welcome for?

**Comment by NickM-27** [comment]
Yes websocket is supported, that is what the frontend uses for controlling and receiving data from frigate

**Reply by billyjacoby** [reply]
Websockets would require having a long lasting open connection between Frigate and an external server though, right?

The end goal would be to have anyone that runs frigate able to use the native app as simply as possible. I was thinking webhooks that get fired on motion events instead of just a websocket connection, but maybe there's an easier was to use websockets to achieve this?

**Reply by NickM-27** [reply]
The web ui just makes a connection whenever it is opened, it's not staying connected in the background

**Reply by billyjacoby** [reply]
Right, that's why I don't think a websocket would work for this use case. 

Instead there would need to be a way for the Frigate server to send some sort of event to an external server all the time, even if not connected to the UI. 

Basically the way that MQTT works now. I also looked into just accepting MQTT events, but it looks like you can only connect a single broker currently. So you wouldn't be able to send events to Homeassistant and an external server.

**Reply by NickM-27** [reply]
Ah gotcha, I don't see why it couldn't be long lived connection.

In any case, there are MQTT proxies that could proxy messages between multiple brokers

**Reply by billyjacoby** [reply]
I'm not sure, but my assumption is that with more than a handful of users long lived websocket connections could get expensive. But i could be wrong, i'll check it out!

And i briefly looked into the MQTT proxy thing. That might be the route that i go to start, but would be less ideal since it's an extra setup step for users. 

I think hooking into the event dispatcher that's currently used for MQTT events to fire a POST request to an endpoint defined in the configuration would be the lowest friction for users, i'm just not nearly as familiar with the backend here as i am with the frontend so it would take a bit of investigating on my end. 

If that's a PR that would be considered it might be worth it though.
